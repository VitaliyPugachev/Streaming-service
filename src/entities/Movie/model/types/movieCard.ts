

export interface MovieCardSchema {
    id: string;
    title: string;
    poster?: string;
    year?: string;
    genre?: string[];
    director?: string;
    country?: string[];
    actors?: string[];
    description?: string;
    rating?: string;
}
