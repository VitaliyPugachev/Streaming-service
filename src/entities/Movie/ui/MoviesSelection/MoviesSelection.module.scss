.MoviesSelection {
    display: flex;
    gap: 10px;
    flex-wrap: wrap;
    padding: 10px 20px;
    width: 100%;
}

.title {
    margin: 2rem 0;
    text-align: center;

}

.container {

}

.section {
    padding: 0 30px;
}
