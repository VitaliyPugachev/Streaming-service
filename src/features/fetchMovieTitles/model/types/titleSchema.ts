export interface TitleSchema {
    title: string,
    id: string,
}

export interface MovieTitles {
    titles: TitleSchema[] | undefined
}
