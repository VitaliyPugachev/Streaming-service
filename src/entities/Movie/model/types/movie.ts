import {MovieCardSchema} from "./movieCard";

export interface Movie {
    isLoading?: boolean;
    error?: string;
    data?: MovieCardSchema
}
