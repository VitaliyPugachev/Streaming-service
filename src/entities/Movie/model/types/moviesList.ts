import {MovieCardSchema} from "./movieCard";

export interface MoviesList {
    isLoading?: boolean;
    error?: string;
    data?: MovieCardSchema[];
}
