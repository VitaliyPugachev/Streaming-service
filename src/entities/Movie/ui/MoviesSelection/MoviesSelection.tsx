import cls from './MoviesSelection.module.scss';
import {MoviesList} from "../../model/types/moviesList";
import {MovieCardSchema} from "../../model/types/movieCard";
import {MovieSectionItem} from "../MovieSectionItem/MovieSectionItem";
import {Loader} from "../../../../shared/ui/Loader/Loader";

interface MoviesSelectionProps {
    moviesList?: MovieCardSchema[];
    isLoading?: boolean;
    error?: string;
}

export const MoviesSelection = (props: MoviesSelectionProps) => {
    const {
        moviesList,
        isLoading,
        error
    } = props;

    if (!moviesList) {
        return (
            <div>
                Not Found
            </div>
        )
    }

    if (isLoading) {
        return (
            <Loader/>
        )
    }


    return (
        <div className={cls.MoviesSelection}>
            {moviesList.map((movie) => (
                <MovieSectionItem movie={movie}/>
            ))}
        </div>
    )
}
