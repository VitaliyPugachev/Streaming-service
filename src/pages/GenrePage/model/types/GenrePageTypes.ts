import {MoviesList} from "../../../../entities/Movie/model/types/moviesList";

export interface GenrePage {
    isLoading: boolean,
    error: string,
    data: MoviesList
}
