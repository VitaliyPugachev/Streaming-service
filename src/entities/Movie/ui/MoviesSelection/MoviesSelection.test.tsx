import {render, screen} from "@testing-library/react";
import {MoviesSelection} from "./MoviesSelection";
import {MovieCardSchema} from "../../model/types/movieCard";

jest.mock('react-router-dom');

describe('MoviesSelection', () => {
    const moviesList: MovieCardSchema[] = [
        {
            id: '1',
            title: 'movie1',
            poster: '',
            description: 'description',
            year: '2023',
            director: 'Director',
            genre: ['adventure'],
            country: ['USA'],
            actors: ['actor'],
            rating: '7'
        },
        {
            id: '2',
            title: 'movie2',
            poster: '',
            description: 'description',
            year: '2023',
            director: 'Director',
            genre: ['adventure'],
            country: ['USA'],
            actors: ['actor'],
            rating: '7'
        },
    ];



    test('element had been rendered in DOM, ui didnt change', () => {
        render(<MoviesSelection title={'popular'} isLoading={false} moviesList={moviesList}/>);
        const element = screen.getByTestId('movieSelection');
        expect(element).toBeInTheDocument();
        expect(element).toMatchSnapshot();
    });
})
