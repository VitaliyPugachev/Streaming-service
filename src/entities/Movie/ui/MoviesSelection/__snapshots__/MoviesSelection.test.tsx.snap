// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesSelection visualization is not changed 1`] = `
<section
  data-testid="movieSelection"
>
  <title
    class="title"
  >
    popular
  </title>
  <div
    class="MoviesSelection"
  />
</section>
`;
